### PR TITLE
Adds initial files with basic funcitonality

### PR DESCRIPTION
**Pull request by aniketsharma00411**
Basic functions for heterovector are added.

cmake files are also added to compile and run example files.﻿

### CODE REVIEW

File: include/heterovector.hpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#ifndef _Matrix_hpp_
```
Comment (ashok-arora) [inline include/heterovector.hpp]: Change this to heterovector

**Review by ashok-arora — CHANGES_REQUESTED** [review]

File: include/heterovector.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            counter++;
+
+    return counter;
+}
+
+int heterovector::size() { return vec.size(); }
+
+template <typename T>
+T heterovector::get(int index) {
+    return std::get<T>(vec[index]);
```
Comment (ashok-arora) [inline include/heterovector.hpp]: std::get can throw std::bad_variant_access exception, consider using a try catch block.

**Review by ashok-arora — CHANGES_REQUESTED** [review]

File: include/heterovector.hpp (hunk 1 above)
Comment (aniketsharma00411) [reply include/heterovector.hpp]: Sure 😆

**Comment by aniketsharma00411** [comment]
I have done the changes. Check again

**Review by ashok-arora — APPROVED** [review]
Looks good to me.

**Pull request merged**